tests/src/application/controllers/change-password.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { type ChangePassword } from '@/domain/use-cases'
+import { type PasswordValidator } from '@/application/validation'
+import { ChangePasswordController } from '@/application/controllers'
+import { MissingParamError, NotFoundError, PasswordMismatchError, PasswordRequirementsError } from '@/application/errors'
+import { badRequest, success } from '@/application/helpers'
+import { makeFakeChangePasswordRequest, makeFakeChangePasswordUseCase, makeFakeServerError, makePasswordValidator } from '@/tests/utils'
+
+interface SutTypes {
+  sut: ChangePasswordController
+  changePasswordStub: ChangePassword
+  passwordValidatorStub: PasswordValidator
+}
+
+const makeSut = (): SutTypes => {
+  const changePasswordStub = makeFakeChangePasswordUseCase()
+  const passwordValidatorStub = makePasswordValidator()
+  const dependencies: ChangePasswordController.Dependencies = {
+    changePassword: changePasswordStub,
+    passwordValidator: passwordValidatorStub
+  }
+  const sut = new ChangePasswordController(dependencies)
+  return {
+    sut,
+    changePasswordStub,
+    passwordValidatorStub
+  }
+}
+
+describe('ChangePasswordController', () => {
+  describe('test the id field', () => {
+    it('Should return 400 if no userId is provided', async () => {
```

Comment:
This message "Should return 400" can be change to "bad request", because we refactor expects to this logic and 400 means "bad request". Make sense?